cloud/pod_util.go

```diff
@@ -281,19 +267,27 @@ func exportECSPodExecutionOptions(ecsConfig evergreen.ECSConfig, containerOpts p
 		windowsVersionConstraint := fmt.Sprintf("%s == %s", ecsWindowsVersionTagConstraint, containerOpts.WindowsVersion)
 		placementOpts.AddInstanceFilters(windowsVersionConstraint)
 	}
-	if arch := exportECSPodArch(containerOpts.Arch); arch != "" {
-		archConstraint := fmt.Sprintf("%s == %s", ecsCPUArchConstraint, arch)
-		placementOpts.AddInstanceFilters(archConstraint)
-	}
 	opts.SetPlacementOptions(*placementOpts)
 
+	var foundCapacityProvider bool
+	for _, cp := range ecsConfig.CapacityProviders {
+		if containerOpts.OS.Matches(cp.OS) && containerOpts.Arch.Matches(cp.Arch) {
```

Comment:
If you only plan to use these two `Matches` in conjunction with one another you could consider merging them into one function and using a bool function on this line.